osutil/stat.go

```diff
@@ -87,3 +87,31 @@ func IsWritable(path string) bool {
 	err := syscall.Access(path, W_OK)
 	return err == nil
 }
+
+// IsDirNotExist tells you whether the given error is due to a directory not existing.
+func IsDirNotExist(err error) bool {
+	switch pe := err.(type) {
+	case nil:
+		return false
+	case *os.PathError:
+		err = pe.Err
+	case *os.LinkError:
+		err = pe.Err
+	case *os.SyscallError:
+		err = pe.Err
+	}
+
+	return err == syscall.ENOTDIR || err == syscall.ENOENT || err == os.ErrNotExist
+}
+
+// DirExists checks whether a given path exists, and if so whether it is a directory.
+func DirExists(fn string) (exists bool, isDir bool, err error) {
```

Comment:
Personally I find this a bit confusing though I also see why you may need something like it.

Reply:
With a little consistency in its use, it becomes less confusing. Not ideal I know, but the stat_t juggling is too tricky to do it every time IMHO.